docs-markdown/src/test/suite/controllers/list-controller.test.ts

```diff
@@ -0,0 +1,123 @@
+import * as chai from 'chai';
+import * as spies from 'chai-spies';
+import { resolve } from 'path';
+import { commands, window } from 'vscode';
+import {
+	automaticList,
+	insertBulletedList,
+	insertListsCommands,
+	insertNestedList,
+	insertNumberedList,
+	removeNestedList
+} from '../../../controllers/list-controller';
+import * as common from '../../../helper/common';
+import * as list from '../../../helper/list';
+import * as telemetry from '../../../helper/telemetry';
+import { loadDocumentAndGetItReady, sleep } from '../../test.common/common';
+
+chai.use(spies);
+
+// tslint:disable-next-line: no-var-requires
+import sinon = require('sinon');
+
+const expect = chai.expect;
+
+suite('List Controller', () => {
+	// Reset and tear down the spies
+	teardown(() => {
+		chai.spy.restore(common);
+	});
+	suiteTeardown(async () => {
+		await commands.executeCommand('workbench.action.closeAllEditors');
+	});
+
+	test('insertListsCommands', () => {
+		const controllerCommands = [
+			{ command: automaticList.name, callback: automaticList },
+			{ command: insertBulletedList.name, callback: insertBulletedList },
+			{ command: insertNestedList.name, callback: insertNestedList },
+			{ command: insertNumberedList.name, callback: insertNumberedList },
+			{ command: removeNestedList.name, callback: removeNestedList }
+		];
+		expect(insertListsCommands()).to.deep.equal(controllerCommands);
+	});
+	test('noActiveEditorMessage', () => {
+		const spy = chai.spy.on(common, 'noActiveEditorMessage');
+		insertNumberedList();
+		expect(spy).to.have.been.called();
+	});
+	test('isMarkdownFileCheck', async () => {
+		const filePath = resolve(__dirname, '../../../../../src/test/data/repo/articles/list.md');
+		await loadDocumentAndGetItReady(filePath);
+		const stub = sinon.stub(telemetry, 'sendTelemetryData');
+		const spy = chai.spy.on(common, 'isMarkdownFileCheck');
+		insertNumberedList();
+		await sleep(100);
+		expect(spy).to.have.been.called();
+		stub.restore();
+	});
+	test('insertList', async () => {
+		const editor = window.activeTextEditor;
+		common.setCursorPosition(editor, 27, 0);
+		await sleep(100);
+		window.showQuickPick = (items: string[] | Thenable<string[]>) => {
```

Comment:
We actually need to stub showQuickPick or else it will interfere with other tests and set the options to your current quickPick for later tests, which causes test failures. See here for an example: https://github.com/microsoft/vscode-docs-authoring/blob/d9f16d58c84427179d24efde462ffac9658a03f4/docs-markdown/src/test/suite/controllers/image-controller.test.ts#L56